### ISSUE TITLE
Unable to run "webui.cmd" from /stable-diffusion-main

### ISSUE THREAD

**Issue by BlackCube008** [hlky/stable-diffusion-webui#145]
Hi,

This is the first time I use github and also the first time I dive deeper into coding. I am not even sure this is the right section to post for help.

I am trying to install Stable Diffusion Webui and am following this guide: [https://rentry.org/GUItard](https://rentry.org/GUItard)
Step 6 suggests running the .cmd file but I don't know how to do it. Bear with me for the question.
Do I need to use Terminal? Because if that is the case, I have moved the file onto Terminal UI and it says:

"zsh: permission denied: /Users/il.Mister/Downloads/stable-diffusion-main/webui.cmd"

Can anyone help?

**Comment by hlky** [comment]
webui.cmd is for windows

On a mac you will need to install conda, then do 
`conda env create -f environment.yaml`
then 
`conda activate <name of environment from environment.yaml>`
finally,
`python scripts\webui.py`
or 
`python scripts\relauncher.py`

You can use .sh files on mac, right?

Try one with contents

```
conda env create -f environment.yaml
conda env update --file environment.yaml --prune
conda activate <name of environment from environment.yaml>
python scripts\relauncher.py
```

this should be the equivalent to webui.cmd on mac and linux too

**Comment by hlky** [comment]
This should help with Apple Silicon support https://github.com/magnusviri/stable-diffusion

**Comment by BlackCube008** [comment]
> `conda env create -f environment.yaml`

Hey thanks for this. I have some difficulties. Here is what I have done:
1. I have duplicated a file that ended with .sh
2. I have changed the content with the below and renamed the file webui.sh
3. I have placed that file on the location stable-diffusion-main
4. I have opened Terminal and have then written sh [space] then I have moved the filw webui.sh onto terminal to start the process
5. I have encountered an error

What am I doing wrong?
What should I write instead of activate <name of environment from environment.yaml>?

---

conda env create -f /Users/[name_of_the_user]/Downloads/stable-diffusion-main/environment.yaml
conda env update --file environment.yaml --prune
conda activate /Users/[name_of_the_user]/Downloads/stable-diffusion-main/environment.yaml
python scripts\relauncher.py
